### PR TITLE
Add Document Export

### PR DESCRIPTION
**Pull request by valentinbootz**
# Document Export

## :recycle: Current situation & Problem

[comment]: # "
*Describe the current situation (if possible with and exemplary (or real) code snippet and/or where this is used)*
*Please link any open issue that is addressed with this PR*
"
~~Creating a document to store knowledge on a web service and exporting it in a local directory or exposing a new endpoint is a common Apodini use case yet currently implementations need a dependency on the ApodiniMigrator package.~~
``ApodiniDocumentExport`` provides a package to create a document to store knowledge on a web service and export it in a local directory or expose a new endpoint. The related PR changes the ``ApodiniMigrator`` API and requires refactoring in ``ApodiniMigration``.

## :bulb: Proposed solution

[comment]: # "
*Describe the solution and how this affects the project and internal structure*
"
~~This PR adds the ApodiniMigrator functionality to support this use case as a new target in the Apodini package.~~
This PR upgrades ``ApodiniMigrator`` version and adds ``ApodiniDocumentExport``dependency to refactor ``ApodiniMigration``.

## :gear: Release Notes 

[comment]: # "
*Add a short summary of the feature as well as possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented if it appends or changes the public interface.*
"
- Update ApodiniMigrator and ApodiniDocumentExport dependencies to refactor ApodiniMigration

## :heavy_plus_sign: Additional Information

[comment]: # "
*Provide some additional information if possible*
"

### Related PRs

[comment]: # "
*Reference the related PRs*
"
[#13](https://github.com/Apodini/ApodiniMigrator/pull/13)

### Testing

[comment]: # "
*Are there tests included? If yes, which situations are tested and which corner cases are missing?*
"
- Testing should not be affected by this PR

### Reviewer Nudging

[comment]: # "
*Where should the reviewer start, where is a good entry point?*
"
Take a look at the related PR to see changes in ``ApodiniMigrator``

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/Apodini/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Apodini/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/Apodini/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Apodini/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by valentinbootz** [comment]
Document export as a feature in the Apodini package introduces a cyclic dependency in the ApodiniMigrator package. @PSchmiedmayer and I decided to create ApodiniDocumentExport as a separate package and refactor ApodiniMigrator to solve this. This PR now instead just refactors ApodiniMigration to use the new API.

**Comment by valentinbootz** [comment]
@PSchmiedmayer Should we create a new release in ApodiniMigrator to use here?

File: Package.swift [hunk 1]
```diff
@@ -114,13 +114,16 @@ let package = Package(
         .package(url: "https://github.com/slashmo/opentelemetry-swift.git", .upToNextMinor(from: "0.1.1")),
         
         // Apodini Migrator
-        .package(url: "https://github.com/Apodini/ApodiniMigrator.git", .upToNextMinor(from: "0.2.0")),
+        .package(url: "https://github.com/Apodini/ApodiniMigrator.git", .branch("develop")),
```
Comment (bauer-andreas) [inline Package.swift]: Created the 0.2.2 release: https://github.com/Apodini/ApodiniMigrator/releases/tag/0.2.2

```suggestion
        .package(url: "https://github.com/Apodini/ApodiniMigrator.git", .upToNextMinor(from: "0.2.2")),
```

**Review by bauer-andreas — APPROVED** [review]
Looks good 👍

File: Package.swift (hunk 1 above)
Comment (valentinbootz) [reply Package.swift]: Great! Thank you 🚀

**Review by PSchmiedmayer — APPROVED** [review]
Thanks for the review @Supereg and for tagging a release in the ApodiniMigrator repo 🚀💪
